Migrate from Vercel Storage to Supabase Storage

toyamarinyon opened the pull request:
We will migrate our application storage from Vercel Storage to Supabase Storage.

## Context
We need reliable file storage for managing Generation Node states and workflow execution states. The application requires immediate access to updated files after write operations. However, with Vercel Storage, we've been experiencing frequent issues where updated file content isn't retrievable even after disabling caching.

I've wrote more details to [docs/adr/0001-migrate-from-vercel-storage-to-supabase-storage.md](https://github.com/giselles-ai/giselle/blob/e710fd2004aa11e4ce94ccfb3ae4b64f2d52d149/docs/adr/0001-migrate-from-vercel-storage-to-supabase-storage.md)

## Before merge 

- [x] Create Supabase Strorage for the production environment and set these environment variables to Vercel Project
- [x] Run the migrate script for the production environment

shige commented on apps/studio.giselles.ai/app/giselle-engine.ts:
```diff
@@ -5,15 +5,18 @@ import { WorkspaceId } from "@giselle-sdk/data-type";
+				accessKeyId: process.env.SUPABASE_ACCESS_KEY_ID ?? "",
+				secretAccessKey: process.env.SUPABASE_SECRET_ACCESS_KEY ?? "",
+				endpoint: process.env.SUPABASE_ENDPOINT ?? "",
+				region: "us-west-1",
```
```suggestion
				region: "us-east-1",
```

shige replied:
Since we have unified the region of Vercel Functions and Vercel Postgres, Supabase Auth to `us-east-1`, we would like you to change it to `us-east-1` as well from a latency perspective.

shige replied:
I updated the suggested change by https://github.com/giselles-ai/giselle/commit/1b9485cc76bdebeb18c30e351f50a517e668910a!
@toyamarinyon

shige reviewed (changes requested):
Could you please just update the region?

toyamarinyon commented:
I've already updated https://github.com/giselles-ai/giselle/pull/684/commits/1b9485cc76bdebeb18c30e351f50a517e668910a

satococoa commented on pnpm-workspace.yaml:
```diff
@@ -78,3 +78,4 @@ catalog:
   "@fal-ai/client": 1.3.0
   langfuse: 3.37.0
   langfuse-core: 3.37.0
+  aws4fetch: 1.0.20
```
:memo: `aws4fetch` is required as a peer dependency.
https://unstorage.unjs.io/drivers/s3

shige reviewed (approved):
Thank you! 👍

satococoa reviewed (approved):
Great :+1:

toyamarinyon commented:
Thanks for the review!  I've discovered and addressed an issue with Supabase Storage compatibility when using the official Unstorage S3-compatible driver.  Specifically, file listing was failing, which broke the file copying functionality.

This impacts the following:

*   **Sample App Creation:**  The sample app created during account creation relies on file copying.
*   **Workflow Preview:**  Workflows containing files (images, PDFs, text, etc.) use file copying during Preview execution.

To resolve this, I've implemented a custom Unstorage driver for Supabase Storage that fixes the file listing problem.

**Please verify the following:**

*   **Account Creation & Sample App:**  Confirm that you can create a new account and that the sample app is successfully created and runs without errors.
*   **Workflow Preview with Files:**  Create a simple workflow that includes a file (any type: image, PDF, text) and verify that it executes correctly in the Preview environment.  The file should be copied as part of the workflow execution.

Let me know if you have any questions or encounter any issues during verification!

shige commented:
📝 I will do my OATs in `preview` branch.

shige reviewed (approved):
Thank you! 👍

shige commented:
> ## Before merge
> * [x]  Create Supabase Strorage for the production environment and set these environment variables to Vercel Project
> * [x]  Run the migrate script for the production environment

I did it!
